dask/array/backends.py

```diff
@@ -296,3 +299,105 @@ def _nannumel_sparse(x, **kwargs):
     # If all dimensions are contracted, this will just be a number, otherwise we
     # want to densify it.
     return n.todense() if hasattr(n, "todense") else n
+
+
+class ArrayBackendEntrypoint(DaskBackendEntrypoint):
+    def __init__(self):
+        """Register data-directed dispatch functions"""
+        raise NotImplementedError
+
+    def default_random_state(self):
+        raise NotImplementedError
+
+    def new_random_state(self, state):
+        raise NotImplementedError
+
+    def ones(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def zeros(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def empty(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def full(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def arange(self, *args, **kwargs):
+        raise NotImplementedError
+
+
+array_creation_dispatch = CreationDispatch(
+    config_field="array.backend.library",
+    default="numpy",
+    name="array_creation_dispatch",
+)
+
+
+class NumpyBackendEntrypoint(DaskBackendEntrypoint):
+    def __init__(self):
+        # Importing this class will already guarentee
+        # that data-dispatch functions are registered
+        pass
+
+    def default_random_state(self):
+        if not hasattr(self, "_np_random_states"):
+            self._np_random_states = da.random.RandomState()
+        return self._np_random_states
+
+    def new_random_state(self, state):
+        return state
```

Comment:
Just flagging that we may want to resolve this with some in-flight work on #9038. That's on me to review, with apologies to the author.

Reply:
Would your suggestion be to leave out random-number generation from the set of "dispatchable" Array-creation funcitons for now? (Note that we certainly don't need to expose **all** possible functions on the first pass)

Reply:
Ah, I see - I just looked at #9038. It looks like these PRs are **mostly** independent, but there is a bit of overlap. (So, I suppose one of us will need to deal with merge conflicts).  The critical change for this PR is that we cannot initalize/use a numpy-based `RandomState` object at import time, because the user may need to specify a non-default array backend later on.

It looks like we should be able to do this for `Generator` as well as `RandomState`.

Reply:
Yeah, I'm not too worried about it, just flagging that we may need to do some light thinking down the line :)